### ISSUE TITLE
Unnecessary call to `bytes`Ruff(UP018) fixing false positive

### ISSUE THREAD

**Issue by sbdchd** [charliermarsh/ruff#1615]
when hovering over the code the LSP warns:

```
Unnecessary call to `bytes` Ruff(UP018)
```

and auto fixing changes

```python
foo = str(b'{"foo":"bar"}')
```

to:

```python
foo = b'{"foo":"bar"}'
```

which has a different type

**Comment by charliermarsh** [comment]
Thanks.
